code/addons/test/src/node/boot-test-runner.ts

```diff
@@ -67,7 +67,10 @@ const bootTestRunner = async (channel: Channel) => {
 
   const startChildProcess = () =>
     new Promise<void>((resolve, reject) => {
-      child = execaNode(vitestModulePath);
+      child = execaNode(vitestModulePath, {
+        env: { VITEST: 'true', TEST: 'true', NODE_ENV: process.env.NODE_ENV ?? 'test' },
```

Comment:
these were already set at runtime in the vitest module, but setting them immediately here means that they will be set before anything else.